src/main/java/io/vertx/codegen/type/TypeReflectionFactory.java

```diff
@@ -73,8 +73,9 @@ public static TypeInfo create(Type type) {
           if (kind == ClassKind.API) {
             java.lang.reflect.TypeVariable<Class<Handler>> classTypeVariable = Handler.class.getTypeParameters()[0];
             Type handlerArg = Helper.resolveTypeParameter(type, classTypeVariable);
-            return new ApiTypeInfo(fqcn, true, typeParams, handlerArg != null ? create(handlerArg) : null, module, false, false);
-          } else if (kind == ClassKind.DATA_OBJECT) {
+            return new ApiTypeInfo(fqcn, true, typeParams, handlerArg != null ? create(handlerArg) : null, module, false, false, null);
```

Comment:
Same here for tests

Reply:
this is on purpose for now, there is no API yet as we have not defined whether we want to annotate `@VertxGen` with `@DataObject` or use a property declaration file. This will be done after. This only change the internal to make this possible.